src/ray/pubsub/publisher.h

```diff
@@ -156,11 +167,12 @@ class PublisherInterface {
   ///
   /// \param channel_type The type of the channel.
   /// \param subscriber_id The node id of the subscriber.
-  /// \param key_id The key_id that the subscriber is subscribing to.
+  /// \param key_id The key_id that the subscriber is subscribing to. std::nullopt if
+  /// subscribing to all.
   /// \return True if registration is new. False otherwise.
   virtual bool RegisterSubscription(const rpc::ChannelType channel_type,
                                     const SubscriberID &subscriber_id,
-                                    const std::string &key_id) = 0;
+                                    const std::optional<std::string> &key_id) = 0;
```

Comment:
nit: will pass a string trigger a constructing a value?

Reply:
Yes this is a copy. It should be ok since the id is usually small, but I can change this to std::string* if you really prefer.

Reply:
I'm ok with this since it's not a big issue here. Just want to clarify this.